src/util/schnorr.rs

```diff
@@ -52,29 +52,44 @@ impl str::FromStr for TweakedPublicKey {
     }
 }
 
+/// Untweaked BIP-340 key pair
+pub type UntweakedKeyPair = KeyPair;
+
+/// Tweaked BIP-340 key pair
+#[derive(Clone, Debug)]
+pub struct TweakedKeyPair(KeyPair);
+
 /// A trait for tweaking Schnorr public keys
 pub trait TapTweak {
-    /// Tweaks an untweaked public key given an untweaked key and optional script tree merkle root.
-    ///
-    /// This is done by using the equation Q = P + H(P|c)G, where
-    ///  * Q is the tweaked key
-    ///  * P is the internal key
+    /// Tweaked key type with optional auxiliary information
```

Comment:
In d2cdbddd0b57452f6b0564f1ce0da3b39a684a34
nit: The documentation for the trait should be updated here to also state it allows keypair tweaking, The description of the tweaking can also be updated with one line stating that for keypairs the secret keys inside the keypair is also tweaked.